src/services/preProcess.ts

```diff
@@ -345,6 +345,42 @@ namespace ts {
                     break;
                 }
 
+                if (scanner.getToken() === SyntaxKind.TemplateHead) {
+                    const stack = [scanner.getToken()];
+                    loop: while (true) {

```

Comment:
Shouldn't this loop be `while (length(stack))`? Otherwise is there a chance that you just start losing import statements?

```ts
let x = `hello ${await import("module").default}`;

import { y } from "some-other-module";
```

Reply:
Thanks for the review.

Reply:
@DanielRosenwasser I think the example works correctly for both cases `while (true) {` and `while(length(stack))`. Anyway, I changed to `while (length(stack))`, it looks safer.

Reply:
@DanielRosenwasser Do you have any concerns?